### PR TITLE
Fix Cisco ASA HA state sensors - Issue ID #16544 

### PR DESCRIPTION
**Pull request by rudybroersma**
This fixes 16544 for Cisco ASA HA state sensors. Prior to this patch, running discovery.php on a Cisco ASA would reset cfwHardwareStatus sensors:

afbeelding

Prior to running discovery:

    MariaDB [librenms]> select sensor_type, sensor_descr, sensor_current from sensors where device_id = 1241 and sensor_class = 'state';
    +-------------------+------------------------------+----------------+
    | sensor_type       | sensor_descr                 | sensor_current |
    +-------------------+------------------------------+----------------+
    | cfwHardwareStatus | Failover LAN Interface       |              2 |
    | cfwHardwareStatus | Primary Unit                 |              9 |
    | cfwHardwareStatus | Secondary Unit (this Device) |             10 |
    +-------------------+------------------------------+----------------+
    3 rows in set (0.000 sec)

After running discovery:

    MariaDB [librenms]> select sensor_type, sensor_descr, sensor_current from sensors where device_id = 1241 and sensor_class = 'state';
    +-------------------+------------------------------+----------------+
    | sensor_type       | sensor_descr                 | sensor_current |
    +-------------------+------------------------------+----------------+
    | cfwHardwareStatus | Failover LAN Interface       |              0 |
    | cfwHardwareStatus | Primary Unit                 |              0 |
    | cfwHardwareStatus | Secondary Unit (this Device) |              0 |
    +-------------------+------------------------------+----------------+
    3 rows in set (0.000 sec)

The sensors were reset to 0/null by the discovery process. The reason for this was that the discovery code was trying to set a string ('standby'  or 'active') as sensor value, which is not allowed. 

This patch adds a lookup table to convert the string value to an integer.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/discovery/sensors/state/asa.inc.php [hunk 1]
```diff
@@ -31,20 +39,37 @@
         ];
         create_state_index($state_name, $states);
 
+        $stateLookupTable = [
```
Comment (murrant) [inline includes/discovery/sensors/state/asa.inc.php:42]: There is no reason to have a lookup table.  Just change the snmp query to give numeric values.

Best to convert it to SnmpQuery if you can too.

Comment (rudybroersma) [reply includes/discovery/sensors/state/asa.inc.php]: Fixed both!

File: includes/discovery/sensors/state/asa.inc.php [hunk 2]
```diff
@@ -32,19 +40,23 @@
...
+                $oid_index = 6;
             } elseif ($index == 'secondaryUnit') {
-                $index = 7;
+                $oid_index = 7;
             }
+
+            $sensor_value = $temp[$index]['CISCO-FIREWALL-MIB::cfwHardwareStatusValue'];
+
             //Discover Sensors
-            discover_sensor(null, 'state', $device, $cur_oid . $index, $index, $state_name, $descr, 1, 1, null, null, null, null, $temp[$index][' cfwHardwareStatusValue'], 'snmp', $index);
```
Comment (laf) [inline includes/discovery/sensors/state/asa.inc.php:44]: I wonder if the issue was purely the space in the variable being used here.

At least the code's cleaned up a bit so thanks @rudybroersma

**Review by laf — APPROVED** [review]

File: includes/discovery/sensors/state/asa.inc.php (hunk 2 above)
Comment (rudybroersma) [reply includes/discovery/sensors/state/asa.inc.php:44]: Yeah in retrospect I wondered the same. At least it's now migrated to SnmpQuery :)

**Pull request merged**